logs_analyzer/app/scala/src/main/scala/com/databricks/apps/logs/ApacheAccessLog.scala

```diff
@@ -0,0 +1,42 @@
+package com.databricks.apps.logs
+
+/**
+  * An entry of Apache access log.
+  */
+
+case class ApacheAccessLog(ipAddress: String,
+                           clientIdentd: String,
+                           userId: String,
+                           dateTime: String,
+                           method: String,
+                           endpoint: String,
+                           protocol: String,
+                           responseCode: Int,
+                           contentSize: Long) extends Serializable {
+}
+
+object ApacheAccessLog {
+  private val PATTERN = """^(\S+) (\S+) (\S+) \[([\w:/]+\s[+\-]\d{4})\] "(\S+) (\S+) (\S+)" (\d{3}) (\d+)""".r
+
```

Comment:
It is more accurate to say that that Sun style influenced the various Scala styles. [Scalafmt](https://olafurpg.github.io/scalafmt/) is a code formatter for Scala with defaults that can be tuned.